### ISSUE TITLE
check with valgrind

### ISSUE THREAD

**Issue by notestaff** [refresh-bio/KMC#90]
I see various warnings about possible memory corruption when running KMC under valgrind ( http://valgrind.org/ ).   @marekkokot maybe try it?   In my experience valgrind warnings often point to real problems.

**Comment by marekkokot** [comment]
Thanks for the suggestion. I hope I will find the time to check it with Valgrind, but probably it will not be soon (I last used Valgrind a couple years ago and I suspect it will take a significant amount of time to run in and find causes of its warnings).

**Comment by notestaff** [comment]
@marekkokot it’s actually doesn’t take much time.  E.g. it quickly found that in kmer_file.cpp when you fread max_count it should now be sizeof(uint64) not uint32.

**Comment by notestaff** [comment]
(or if have to read a 32-bit value for compatibility, read into a uint32 then assign that to max_count?)

**Comment by notestaff** [comment]
btw maybe mention in the docs that kmc databases are not portable across architectures
https://stackoverflow.com/questions/2193472/are-files-dumped-with-fwrite-portable-across-different-systems

**Comment by marekkokot** [comment]
Hi, thanks for pointing that. In fact we was intended to make it portable, and I believe KMC database format itself was portable. 
We have dedicated functions like: https://github.com/refresh-bio/KMC/blob/master/kmer_counter/kb_completer.cpp#L303
KMC database format was portable, but API was not. 
Probably when implementing new features I I neglected portability calling fwrite with sizeof.
I have created to remember to check this #93
